MaxText/integration/tunix/weight_mapping/llama3.py

```diff
@@ -0,0 +1,97 @@
+from dataclasses import dataclass
+import numpy as np
+import jax
+
+
+@dataclass
+class LLAMA3_VLLM_MAPPING:
+  """Mapping MaxText Llama 2 and Llama 3 weights to vLLM's Llama 2 and Llama 3 weights."""
+
+  def to_hf_hook_fns():
+
+    def reorder_rope(arr):
+      evens = arr[..., ::2]
+      odds = arr[..., 1::2]
+      return jax.numpy.concatenate((evens, odds), axis=arr.ndim - 1)
+
+    def transform_query_kernel(arr):
+      head_dim = arr.shape[-1]
+      depth_scale = np.dtype("float32").type(np.sqrt(head_dim))
+      arr = arr * depth_scale
+      return reorder_rope(arr)
+
+    hook_fns = {
+        "base.decoder.layers.self_attention.query.kernel": transform_query_kernel,
+        "base.decoder.layers.self_attention.key.kernel": reorder_rope,
```

Comment:
@wenxindongwork I recall that this is only needed for Llama2/Llama3 not Llama3.1/Llama3 (or the other way around), perhaps we need to unify it for MaxText.

@A9isha 
Maybe this issue was causing convergence issues?

Reply:
Gagik, WenXin and I discussed offline and we are specifying and we support for Llama3.1 only at the moment. Follow up PR to support more models.